src/server/http/index.js

```diff
@@ -7,19 +7,21 @@ import { setupVersionCheck } from './version_check';
 import { handleShortUrlError } from './short_url_error';
 import { shortUrlAssertValid } from './short_url_assert_valid';
 import { shortUrlLookupProvider } from './short_url_lookup';
-import { setupConnection } from './setup_connection';
 import { setupRedirectServer } from './setup_redirect_server';
 import { registerHapiPlugins } from './register_hapi_plugins';
-import { setupBasePathRewrite } from './setup_base_path_rewrite';
 import { setupXsrf } from './xsrf';
 
 export default async function (kbnServer, server, config) {
   server = kbnServer.server = new Hapi.Server();
 
   const shortUrlLookup = shortUrlLookupProvider(server);
 
-  setupConnection(server, config, kbnServer.newPlatform);
-  setupBasePathRewrite(server, config);
+  server.connection({
```

Comment:
__note:__ the rest of properties is configured in `@kbn/platform` that's the bare minimum we should define here.